backend/mtproto/mtproto.go

```
@@ -0,0 +1,655 @@
+func writeFileAtomic(path string, payload []byte) error {
+	dir := filepath.Dir(path)
+	if err := os.MkdirAll(dir, 0o700); err != nil {
+		return err
+	}
+
+	tempFile, err := os.CreateTemp(dir, filepath.Base(path)+".*.tmp")
+	if err != nil {
+		return err
+	}
+	tempPath := tempFile.Name()
+	defer os.Remove(tempPath)
```

golangci-lint (2.12.2)

[error] Error return value of `os.Remove` is not checked

(errcheck)